Standardize renderer names and terms

tetrapod00 opened the pull request:
Part of https://github.com/godotengine/godot-docs/issues/10139. (Doesn't close, because some changes must be made in the class ref still.)
See also https://github.com/godotengine/godot/pull/98747, https://github.com/godotengine/godot/pull/98744.

Standardizes terms for renderers/rendering methods, rendering drivers, and "rendering backends".

New standards, for *user-facing docs*:
- Godot has three *renderers* or *rendering methods*: **Forward+**, **Mobile**, and **Compatibility**. In most user-facing docs, these are referred to as "renderers". They are exposed by the engine as "rendering methods", and "rendering method" is a valid term to use.
- Godot has four *rendering drivers*: **OpenGL**, **Vulkan**, **Direct3D 12**, **Metal**. 
- Godot has a rendering abstraction layer called "RenderingDevice". This can be referred to as the "RenderingDevice backend".
- The term "backend" should be avoided in user-facing docs, as it is ambiguous. Especially the terms "Compatibility backend", "Forward+ backend", "Vulkan backend", are to be avoided.
- The Forward+ renderer is called "Forward+". Avoid "Forward Plus" or "forward_plus" unless `+` is not a valid symbol.
- The Mobile renderer is called "Mobile". Avoid "Forward Mobile". This rendering method is exposed in the project settings as "mobile" and referred to as "Mobile" in the engine interface.
- The Compatibility renderer is called "Compatibility". Only use "GL Compatibility" when needed. For example, `gl_compatibility` is how it is exposed in the project settings.

This PR changes:
- Almost all references to "Forward Mobile" to "Mobile". I left a lot of existing references on the Internal Rendering Architecture page.
- All references to "Clustered Forward" to "Forward+".
- Most ambiguous "backend" references to explicitly refer to renderers or rendering drivers.
- Some references to "rendering methods" to "renderers". "Rendering method" is still valid terminology.
- Copies the renderer description from the Renderers page to the List of Features page.
- Corrects some incorrect information on the Internal Rendering Architecture page.


Since the existing terms are nonstandardized, I probably missed a few instances.

skyace65 reviewed (changes requested):
Looks good! Just needs a few fixes.

tetrapod00 commented on contributing/development/core_and_modules/internal_rendering_architecture.rst:
```diff
@@ -542,13 +541,12 @@ Global illumination
-    VoxelGI and SDFGI are only available in the Forward+ backend, not the
-    Forward Mobile or Compatibility methods.
+    VoxelGI and SDFGI are only available in the Forward+ renderer, not the
+    Forward Mobile or Compatibility renderers.
```
I intentionally left some references to "Forward Mobile" on the Internal Rendering Architecture page alone, because I thought "Forward Mobile" was still a valid name for this renderer, just not the one we should use in most places. For example, the heading on this page is still called "Forward Mobile".

I'm not against fully standardizing, though. If all the references on this page change, we should probably change the descriptions of the renderers in `Renderers` and `List of Features` to not say that Mobile is also called "Forward Mobile".

tetrapod00 replied:
I went ahead with the complete standardization in the manual.

tetrapod00 commented:
Removed all references to "Forward Mobile" from the manual, except for a single note on the Internal rendering architecture page that says "Forward Mobile" is the internal name.
Removed all references to "Clustered Forward" from the manual.

I believe the original issue and the terminology used was approved by Calinou and Clay, but we should wait to merge this until one of them gives it at least a quick look.

AThousandShips commented on tutorials/shaders/advanced_postprocessing.rst:
```diff
@@ -104,14 +104,15 @@ inverse of the projection matrix, which in Godot, is accessible with the variabl
-Reconstruct the NDC using ``SCREEN_UV`` for the ``x`` and ``y`` axis, and
-the depth value for ``z``.
+from ``0.0`` to ``1.0`` in the ``z`` direction when using the Forward+ or Mobile
+renderers. Reconstruct the NDC using ``SCREEN_UV`` for the ``x`` and ``y`` axis,
```
```suggestion
renderers. Reconstruct the NDC using ``SCREEN_UV`` for the ``x`` and ``y`` axes,
```

mhilbrunner reviewed (approved):
This needs a rebase, otherwise looks good to me.

tetrapod00 commented:
Rebased, but this still does need at least a cursory review from someone in rendering

mhilbrunner commented:
cc @Calinou and @clayjohn

skyace65 commented:
Thanks!

tetrapod00 commented:
I don't really think this one is worth the effort to cherrypick to 4.3. Removing the label.
